src/ECModel.jl

```diff
@@ -852,6 +853,187 @@ function split_financial_terms(ECModel::AbstractEC, profit_distribution=nothing)
+    Ann_peak_charges = JuMP.Containers.DenseAxisArray(
+        [get_value(ECModel.results[:C_Peak_tot_us], u)
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+                , year_set, user_set
+    )
+
+    # Get revenes by selling energy and costs by buying or consuming energy
+    zero_if_negative = x->((x>=0) ? x : 0.0)
+    Ann_energy_revenues = JuMP.Containers.DenseAxisArray(
+        [(u in axes(ECModel.results[:R_Energy_us])[1]) ? sum(zero_if_negative.(ECModel.results[:R_Energy_us][u,:])) : 0.0
+            for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set
+    )
+    Ann_energy_costs = JuMP.Containers.DenseAxisArray(
+        [(u in axes(ECModel.results[:R_Energy_us])[1]) ? sum(zero_if_negative.(.-(ECModel.results[:R_Energy_us][u, :]))) : 0.0
+        for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set
+    )
+    Ann_net_energy_costs = Ann_energy_costs .- Ann_energy_revenues
+    
+    # Total OPEX costs
+    OPEX = Ann_Maintenance .+ Ann_peak_charges .+ Ann_net_energy_costs
+
+    # get NPV given the reward allocation
+    #Check how to proceed
+    NPV = profit_distribution
+
+    # Total reward
+    Ann_reward = NPV .+ CAPEX .+ OPEX .+ Ann_Replacement .- Ann_Recovery
+    
+    return (
+        NPV=NPV,
+        CAPEX=CAPEX,
+        OPEX=OPEX,
+        OEM = Ann_Maintenance,
+        REP=Ann_Replacement,
+        RV=Ann_Recovery,
+        REWARD=Ann_reward,
+        PEAK=Ann_peak_charges,
+        EN_SELL=Ann_energy_revenues,
+        EN_CONS=Ann_energy_costs,
+        EN_NET=Ann_net_energy_costs,
+    )
+end
+
+function business_plan_dataframe(ECModel::AbstractEC,profit_distribution=nothing, user_set_financial=nothing)
+    gen_data = ECModel.gen_data
+    
+    project_lifetime = field(gen_data, "project_lifetime")
+    year_set = 1:project_lifetime
+    
+    if isnothing(user_set_financial)
+        user_set_financial = [EC_CODE; get_user_set(ECModel)]
+    end
+    if isnothing(profit_distribution)
+        user_set = get_user_set(ECModel)
+        profit_distribution = JuMP.Containers.DenseAxisArray(
+            [ 0.0
+                for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set,
+        )
+    end
+
+    # Create a vector of years from 2023 to (2023 + project_lifetime)
+    gen_data = ECModel.gen_data
+    project_lifetime = field(gen_data, "project_lifetime")
+    years = 1:project_lifetime
+
+    business_plan = split_yearly_financial_terms(ECModel)
+
+    # Create an empty DataFrame
+    df_business = DataFrame(Year = Int[], CAPEX = Float64[], OEM = Float64[], EN_SELL = Float64[], EN_CONS = EN_SELL = Float64[], REP = Float64[], 
+    REWARD = Float64[], RV = Float64[], PEAK = Float64[])
+    for i in years
+        CAPEX = sum(business_plan.CAPEX[i, :])
+        Year = 2022 + years[i]

```

Comment:
I'd avoid using the 2022 as starting date, I think it is no problem to start from 0